src/js/utils/helpers.tsx

```diff
@@ -208,3 +218,77 @@ export const mergeChangesets = (original: Changeset, adding: Changeset) =>
   mergeWith(cloneDeep(original), adding, (objVal, srcVal) =>
     union(objVal, srcVal),
   );
+
+export const getSchemaForChangeset = (
+  schema: DatasetSchema,
+  changes: Changeset,
+) => {
+  const matchedSchema: DatasetSchema = {};
+
+  for (const [objectName, fieldNames] of Object.entries(changes)) {
+    if (schema[objectName]) {
+      const fields = pick(schema[objectName].fields, fieldNames);
+      matchedSchema[objectName] = { ...schema[objectName], fields };
+    }
+  }
+
+  return matchedSchema;
+};
+
+export const filterChangesetBySchema = (
+  schema: DatasetSchema,
+  changes: Changeset,
+) => {
+  const matchedChangeset: Changeset = {};
+  const unmatchedChangeset: Changeset = {};
+
+  for (const [objectName, fieldNames] of Object.entries(changes)) {
+    if (schema[objectName]) {
+      const matchedFields = intersection(
+        fieldNames,
+        Object.keys(schema[objectName].fields),
+      );
+      if (matchedFields.length) {
+        matchedChangeset[objectName] = matchedFields;
+      }
+      const unmatchedFields = without(fieldNames, ...matchedFields);
+      if (unmatchedFields.length) {
+        unmatchedChangeset[objectName] = unmatchedFields;
+      }
+    } else {
+      unmatchedChangeset[objectName] = fieldNames;
+    }
+  }
+
+  return { matchedChangeset, unmatchedChangeset };
+};
+
+export const filterSchema = (schema: DatasetSchema, search: string) => {
+  if (!search) {
+    return schema;
+  }
+
+  const term = search.toLowerCase();
+  const filteredSchema: DatasetSchema = {};
+
+  for (const [name, value] of Object.entries(schema)) {
+    if (
+      name.toLowerCase().includes(term) ||
+      value.label.toLowerCase().includes(term)
+    ) {
+      filteredSchema[name] = value;
```

Comment:
@davidmreed As this is currently written, when a search term in the filter box matches the name (or label) of an Object, we're showing _all_ Fields contained in that Object (even ones that do not match the search term). Otherwise I can't think of how else searching for an Object name/label would work, if we're always _also_ filtering all Fields. But I wanted to confirm this with you or someone else at SFDO.

![Screen Shot 2022-07-21 at 5 06 42 PM](https://user-images.githubusercontent.com/552316/180315286-ef8b964f-57a9-43a0-bf6d-6cca48879113.jpg)

Reply:
Shane confirmed this is okay: https://oddbird.slack.com/archives/CB2576XT4/p1658507412096779